src/canvas.class.js

```diff
@@ -480,9 +480,10 @@
       var ctx = this.contextCache,
           originalColor = target.selectionBackgroundColor;
 
-      target.hasBorders = target.transparentCorners = false;
       target.selectionBackgroundColor = '';
 
+      this.clearContext(ctx);
```

Comment:
moving this higher in the function makes debugging the result of this function much easier.